moveit_core/robot_state/src/robot_state.cpp

```diff
@@ -2076,6 +2101,56 @@ double RobotState::testAbsoluteJointSpaceJump(const JointModelGroup* group, std:
   return 1.0;
 }
 
+double RobotState::testCartesianSpaceJump(const JointModelGroup* group, const LinkModel* link,
+                                          std::vector<RobotStatePtr>& traj, const MaxEEFStep& max_step)
+{
+  double percentage = 1.0;
+  if (traj.size() <= 1)
+    return percentage;
+
+  Eigen::Affine3d start_pose = traj[0]->getGlobalLinkTransform(link);
+  Eigen::Affine3d mid_pose;
+  Eigen::Affine3d end_pose;
+  Eigen::Quaterniond start_quaternion(start_pose.rotation());
+  Eigen::Quaterniond mid_quaternion;
+  Eigen::Quaterniond end_quaternion;
+
+  std::size_t traj_ix = 1;
+  for (; traj_ix < traj.size(); ++traj_ix)
+  {
+    end_pose = traj[traj_ix]->getGlobalLinkTransform(link);
+
+    RobotState robot_mid_state = RobotState(*traj[traj_ix]);
+    traj[traj_ix]->interpolate(*traj[traj_ix - 1], 0.5, robot_mid_state);
+
+    mid_pose = robot_mid_state.getGlobalLinkTransform(link);
+
+    if (max_step.rotation > 0.0)
+    {
+      mid_quaternion = mid_pose.rotation();
+      end_quaternion = end_pose.rotation();
+      double first_rotation_distance = start_quaternion.angularDistance(mid_quaternion);
+      double second_rotation_distance = end_quaternion.angularDistance(mid_quaternion);
+      if (first_rotation_distance > max_step.rotation || second_rotation_distance > max_step.rotation)
+        break;
+    }
+
+    if (max_step.translation > 0.0)
+    {
+      double first_translation_distance = (start_pose.translation() - mid_pose.translation()).norm();
+      double second_translation_distance = (end_pose.translation() - mid_pose.translation()).norm();
+      if (first_translation_distance > max_step.translation || second_translation_distance > max_step.translation)
+        break;
+    }
+
+    start_pose = end_pose;
+    start_quaternion = end_quaternion;
+  }
+  percentage = (double)(traj_ix) / (double)(traj.size());
+  traj.resize(traj_ix);
```

Comment:
why this resize behavior?

Reply:
This trims the trajectory before the jump if one is found. If no jump is found traj_ix is equal to the size of the trajectory leaving the trajectory unchanged

Reply:
please add this as a comment